app/forms/PropertyDetailsFormsValidation.scala

```diff
@@ -48,21 +46,38 @@ object PropertyDetailsFormsValidation {
         case _ => Seq(None)
       }
     }
+
     isTaxAvoidance match {
       case Some(isAvoidTax) if isAvoidTax => validateAvoidanceScheme(avoidanceSchemeNo) ++ validatePromoterReference(promoterReference)
       case _ => Seq(None)
     }
   }
-
-  def validateBuildDate(periodKey:Int, f: Form[_], isNewBuild: Option[Boolean]): Seq[Option[FormError]] = {
+//TODO this can be sploit
+  def validateBuildDate(periodKey: Int, f: Form[_], isNewBuild: Option[Boolean]): Seq[Option[FormError]] = {
     if (isNewBuild == Some(true))
       validateDate(periodKey, f, "newBuildDate", mustBeInChargeablePeriod = true, isMandatory = true) ++ validateDate(periodKey, f, "localAuthRegDate", mustBeInChargeablePeriod = true, isMandatory = true)
     else
       validateDate(periodKey, f, "notNewBuildDate", isMandatory = true)
   }
 
+  def validatedNewBuildDate(periodKey: Int, f: Form[_]): Seq[Option[FormError]] = {
+    validateDate(periodKey, f, "newBuildOccupyDate", mustBeInChargeablePeriod = true, isMandatory = true) ++
+      validateDate(periodKey, f, "newBuildRegisterDate", mustBeInChargeablePeriod = true, isMandatory = true)
+  }
 
-  def checkRevaluedDate(periodKey: Int, isPropertyRevalued : Option[Boolean], revaluedDate : Option[LocalDate]): Seq[Option[FormError]] = {
+  def validatedAcquisitionDate(periodKey: Int, f: Form[_]): Seq[Option[FormError]] = {
```

Comment:
Since these three functions don't add any errors, I'd suggest removing them and their validation functions that call them to clean code up

Reply:
Yeah I agree.